File: benchmarks/factorial/factorial.py
# factorial.py
# Python implementation of factorial benchmark

# Configuration
N1 = 10
N2 = 12
N3 = 15
N4 = 18

# Markers matching Lamina version
HEADER_MARKER = 123456789
FOOTER_MARKER = 987654321

def factorial_iterative(n):
    """Compute factorial iteratively: n! = n * (n-1) * ... * 2 * 1"""
    if n == 0 or n == 1:
        return 1

    result = 1
    for i in range(2, n + 1):
        result *= i
    return result

def main():
    # Print header marker
    print(HEADER_MARKER)

    # Compute and print factorial values
    print(factorial_iterative(N1))
    print(factorial_iterative(N2))
    print(factorial_iterative(N3))
    print(factorial_iterative(N4))

    # Print footer marker
    print(FOOTER_MARKER)

if __name__ == "__main__":
    main()

File: benchmarks/factorial/factorial.rs
// factorial.rs
// Rust implementation of factorial benchmark

// Configuration
const N1: i32 = 10;
const N2: i32 = 12;
const N3: i32 = 15;
const N4: i32 = 18;

// Markers matching Lamina version
const HEADER_MARKER: i64 = 123456789;
const FOOTER_MARKER: i64 = 987654321;

fn factorial_iterative(n: i32) -> i64 {
    if n == 0 || n == 1 {
        return 1;
    }

    let mut result: i64 = 1;
    for i in 2..=n {
        result = result.saturating_mul(i as i64);
    }
    result
}

fn main() {
    // Print header marker
    println!("{}", HEADER_MARKER);

    // Compute and print factorial values
    println!("{}", factorial_iterative(N1));
    println!("{}", factorial_iterative(N2));
    println!("{}", factorial_iterative(N3));
    println!("{}", factorial_iterative(N4));

    // Print footer marker
    println!("{}", FOOTER_MARKER);
}



